### ISSUE TITLE
Sim_tape broken improper changing of UNIT->dynflags

### ISSUE THREAD

**Issue by rcornwell** [simh/simh#707]
Commit: db26349bd75f2fe796234a3206505e4a14cc661f. break sim_tape.c preventing B5500 simulator from working.

Running xiii_build.ini hangs after this commit. Debug messages show that it is not able to read the tape data correctly.

- #### Context

- #### the output of "sim> SHOW VERSION" while running the simulator which is having the issue
```
B5500 simulator V4.0-0 Current        git commit id: db26349b
sim> show version
B5500 simulator V4.0-0 Current
    Simulator Framework Capabilities:
        64b data
        32b addresses
        no Ethernet
        Idle/Throttling support is available
        Virtual Hard Disk (VHD) support
        RAW disk and CD/DVD ROM support
        Asynchronous I/O support (Lock free asynchronous event queue)
        Asynchronous Clock support
        FrontPanel API Version 12
    Host Platform:
        Compiler: GCC 5.4.0 20160609
        Simulator Compiled as C arch: x64 (Release Build) on Jun  7 2019 at 22:47:30
        Memory Access: Little Endian
        Memory Pointer Size: 64 bits
        Large File (>2GB) support
        SDL Video support: No Video Support
        PCRE RegEx (Version 8.38 2015-11-23) support for EXPECT commands
        OS clock resolution: 1ms
        Time taken by msleep(1): 1ms
        OS: Linux hobbit 4.4.0-142-generic #168-Ubuntu SMP Wed Jan 16 21:00:45 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
        git commit id: db26349b
        git commit time: 2019-05-14T01:04:18-0700
sim> 
```

- #### how you built the simulator or that you're using prebuilt binaries

- #### the simulator configuration file (or commands) which were used when the problem occurred.

Please grab the build job off my web site:

   To get build files please see: http://sky-visions.com/burroughs/quick/files.shtml    

   http://sky-visions.com/burroughs/xiii/kit.zip

- #### the expected behavior and the actual behavior

   When building after 'R DIRECTRY BUILT' in a minute or two you should see: 
R MCP FILE LOADED
R -H/L WITH MCP/DISK MARK XIII MODS RRRRRRRR-
R DKA EU0 SU 1,2,3,4,5, EU1 SU 1,2,3,4,5 WENT READY

   With this commit it just hangs.

- #### you may also need to provide specific pointers to data files that may be necessary to demonstrate the problem

  This problem was detected during routine testing of simulators. It occurs in sims version since I am at current version minus recent timer updates. Note B5500 used P7B tape images.

**Comment by markpizz** [comment]
There was a bug in the reverse read logic in the P7B format code.  It was encountered with one of the .bcd files in your test setup.  I think this should fix things.

From my testing, the TIMER stuff should be sufficiently stable to use.

**Comment by rcornwell** [comment]
Nope I am still getting a return code 4 (Invalid Record Length) from sim_tape_read.

I will test the timer stuff shortly. I will also switch my various sims to using sim_activate_after and sim_timer_activate_time_usecs shortly.

**Comment by markpizz** [comment]
Hmm...
You may want to put this command at the beginning of xiii_build.ini:
    sim> SET ON
This will allow an explicitly entered ^E to exit (return) from the current DO command file without having to process remaining commands.

Meanwhile, back to the point of your problem....

If you type SHOW MT just prior to booting from the card deck, things are as expected with the format as P7B.
If you type SHOW MT after hitting ^E when the card deck has been booted, you see the MT formats have changed to SIMH while the device is still attached.

Nothing in sim_tape will dynamically change the tape format.  In fact, sim_tape_set_fmt() will refuse to change the format of an attached unit.  Meanwhile, the tape format is stored in the UNIT->flags field and format value 0 is SIMH.  If something somewhere in the B5500 simulator is messing with the flag bits containing the tape format then all bets are off.  Actually, due to your concern about how many UNIT->flag bits were going to be available for simulator specific code, I moved the tape format out of the UNIT->flags field and into the UNIT->dynflags.  The UNIT->dynflags field is specific to SCP internal activities and any access or manipulation of that field should ONLY  be done by calls into SCP provided APIs.  I see that b5500_mt.c's mt_reset() routine is explicitly setting the whole value of UNIT->dynflags attempting to mess with tape densities without using a sim_tape API.  This would be a bug in b5500_mt.c that just wasn't noticed previously.

**Comment by markpizz** [comment]
Similar invalid references to UNIT->dynflags are being made in I7000_mt.c.

**Comment by rcornwell** [comment]
Ok... I will try and use the I7000 patch you sent and modify it for B5500 and see if that works.

**Comment by markpizz** [comment]
Fixed by f70875d8b2d6016f828372e713d8ecbba9d6ec7f
